themes/src/main/resources/theme/keycloak-preview/account/resources/app/content/applications-page/ApplicationsPage.tsx

```diff
@@ -15,21 +15,273 @@
+              <Checkbox
+                label="In-use app only"
+                isChecked={false}
+                onChange={this.handleChange}
+                aria-label="In-use app only"
+                id="inUseCheckbox"
+                name="inUseCheckbox"
+              />
+            </ToolbarItem>
+          </ToolbarGroup>
+          <ToolbarSection aria-label="Toolbar Section">
+            <ToolbarGroup>
+              <ToolbarItem>{this.state.applications.length} {this.state.applications.length > 1 ? 'items' : 'item'}</ToolbarItem>
+            </ToolbarGroup>
+            <ToolbarGroup>
+              <ToolbarItem>Active Filters: None</ToolbarItem>
+            </ToolbarGroup>
+          </ToolbarSection>
+        </Toolbar>
+        <DataList aria-label="device-activity">
+          {this.state.applications.map((application: Application, appIndex: number) => {
+            return (
+              <DataListItem key={'application-' + appIndex} aria-labelledby="simple-item2" isExpanded={this.state.isRowOpen[appIndex]}>
+                <DataListItemRow>
+                  <DataListToggle
+                    onClick={() => this.onToggle(appIndex)}
+                    isExpanded={this.state.isRowOpen[appIndex]}
+                    id={'applicationToggle' + appIndex}
+                    aria-controls="ex-expand1"
+                  />
+                  <DataListItemCells
+                    dataListCells={[
+                      <DataListCell key={'app-' + appIndex}>
+                        <span id="check-action-item1">
+                          <BuilderImageIcon/> {application.clientId}
+                          </span>
+                        </DataListCell>,
+                        <DataListCell key={'internal-' + appIndex}>
+                          <span>{application.internal ? 'Internal' : 'Third-party'}</span>
+                        </DataListCell>,
+                        <DataListCell key={'status-' + appIndex}>
+                          <span>{application.inUse ? 'In-use' : 'NaN'}</span>
+                        </DataListCell>,
+                        <DataListCell key={'url-' + appIndex}>
+                          <a href={application.url}> <LinkIcon/> {application.url}</a>
+                        </DataListCell>
+                    ]}
+                  />
+                </DataListItemRow>
+                <DataListContent 
+                  noPadding={false}
+                  aria-label="Application Details"
+                  id="ex-expand1"
+                  isHidden={!this.state.isRowOpen[appIndex]}
+                >
+                  <div className="pf-c-content">
+                    <h3>Client</h3>
+                    <hr/>
+                    <h3>Description</h3>
+                    <h3>URL</h3>
+                    <h3>Has access to</h3>
+                    <h3>Access granted on</h3>
+                    <hr/>
+                    <React.Fragment>
```

Comment:
It's probably better to use PF4 layouts here instead of <div>, <h3> and <hr/> tags